plugin-java/src/main/java/appland/javaAgent/AppMapJavaAgentDownloadService.java

```diff
@@ -0,0 +1,173 @@
+    private static final Logger LOG = Logger.getInstance(AppMapJavaAgentDownloadService.class);
+    private static final String AGENT_LINK_FILENAME = "appmap.jar";
+
+    public static AppMapJavaAgentDownloadService getInstance() {
+        return ApplicationManager.getApplication().getService(AppMapJavaAgentDownloadService.class);
+    }
+
+    /**
+     * @return Local path to the downloaded Java agent file.
+     * {@code null} is returned if the agent has not been downloaded yet.
+     */
+    public @Nullable Path getJavaAgentPathIfExists() {
+        var jarFilePath = getJavaAgentPath();
+        return jarFilePath == null || Files.notExists(jarFilePath) ? null : jarFilePath;
+    }
+
+    @SuppressWarnings("DialogTitleCapitalization")
+    public void downloadJavaAgent(@NotNull Project project) {
+        var title = AppMapBundle.get("javaAgent.download.title");
+        new Task.Backgroundable(project, title, false, PerformInBackgroundOption.ALWAYS_BACKGROUND) {
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                try {
+                    downloadJavaAgentSync(indicator);
+                } catch (IOException e) {
+                    LOG.warn("failed to download AppMap Java agent", e);
+                }
+            }
+        }.queue();
+    }
+
+    public boolean downloadJavaAgentSync(@NotNull ProgressIndicator indicator) throws IOException {
+        var agentFilePath = getJavaAgentPath();
+        if (agentFilePath == null) {
+            LOG.warn("unable to locate target file path for AppMap Java agent");
+            return false;
+        }
+
+        var latestAsset = GitHubRelease.getLatestRelease(indicator, "getappmap", "appmap-java")
+                .stream()
+                .filter(asset -> "application/java-archive".equals(asset.getContentType()))
+                .findFirst()
+                .orElse(null);
+
+        if (latestAsset == null) {
+            LOG.warn("JAR assets not found in latest GitHub release of getappmap/appmap-java");
+            return false;
+        }
+
+        // e.g. ~/.appmap/lib/java/appmap-1.7.2.jar
+        var assetDownloadPath = agentFilePath.resolveSibling(latestAsset.getFileName());
+
+        // don't download if the asset file does already exist
+        if (Files.exists(assetDownloadPath)) {
+            return false;
+        }
+
+        // e.g. appmap-1.7.2.downloading
+        var lockFilePath = assetDownloadPath.resolveSibling(latestAsset.getFileName().replace(".jar", ".downloading"));
+        assert !assetDownloadPath.equals(lockFilePath);
+
+        if (!guardWithLockFile(lockFilePath, () -> downloadAgentJarFile(indicator, latestAsset, assetDownloadPath))) {
+            return false;
+        }
+
+        // create a symbolic link pointing to "appmap.jar"
+        try {
+            Files.deleteIfExists(agentFilePath);
+            Files.createSymbolicLink(agentFilePath, assetDownloadPath);
```

Comment:
This symbolic link points to an absolute file path:

![image](https://github.com/getappmap/appmap-intellij-plugin/assets/45714532/ebe58e29-a04d-48fc-b124-86b3a6dd5186)

Please make it so that it points to a relative path, like this:

![image](https://github.com/getappmap/appmap-intellij-plugin/assets/45714532/057ecd2d-5752-4d5a-a176-f76527ef3260)

Reply:
Fixed by https://github.com/getappmap/appmap-intellij-plugin/pull/300/commits/0ead3e9c961bdf0a97243753352e42839d334b04